common-rendering/src/hooks/useTimeAgo.tsx

```diff
@@ -0,0 +1,50 @@
+import { timeAgo } from "@guardian/libs";
+
+import { useInterval } from "./useInterval";
+
+const DEFAULT_INTERVAL = 15 * 1000; // 15 seconds
+
+type Options = {
+	interval?: number;
+};
+
+const updateTimeElements = () => {
+	document
+		.querySelectorAll("time[data-relativeformat]")
+		.forEach((element) => {
+			if (element instanceof HTMLElement) {
+				// Get required values
+				const { relativeformat: relativeFormat } = element.dataset;
+				const absoluteTime = element.getAttribute("datetime");
+				if (!absoluteTime || !relativeFormat) return [];
+
+				let newTime;
+				switch (relativeFormat) {
+					case "short":
+					case "med":
+						newTime = timeAgo(new Date(absoluteTime).getTime(), {
+							verbose: false,
+						});
+						break;
+					case "long":
+						newTime = timeAgo(new Date(absoluteTime).getTime(), {
+							verbose: true,
+						});
+						break;
+				}
+				const oldTime = element.innerText;
+				if (newTime && newTime !== oldTime) element.innerHTML = newTime;
+			}
+		});
+};
+
+/**
+ * Sets up an interval in which the timeAgo function is run against every time element on the
+ * page which has the data-relativeformat attribute set
+ * @param {Number} interval - The length of time in-between time being updated
+ * */
+export const useTimeAgo = (options?: Options) => {
+	useInterval(() => {
+		updateTimeElements();
+	}, options?.interval || DEFAULT_INTERVAL);
+};
```

Comment:
I wonder if it's worth putting this code (`useTimeAgo.tsx` + `useInterval`) directly into `@guardian/libs`

@SiAdcock @sndrs  I know these are React focussed functions but maybe they're useful in other react codebases too?

Reply:
Especially, the `updateTimeElements` function as that is continuing the contract that exists in Frontend around the `relativeformat` data attribute which I think is important.